src/main/java/org/opensearch/knn/index/util/Faiss.java

```diff
@@ -76,62 +75,111 @@ class Faiss extends NativeLibrary {
                     methodComponentContext
                 ).build())
             )
-            .build(),
-        KNNConstants.ENCODER_PQ,
-        MethodComponent.Builder.builder(KNNConstants.ENCODER_PQ)
-            .addParameter(
-                ENCODER_PARAMETER_PQ_M,
-                new Parameter.IntegerParameter(
-                    ENCODER_PARAMETER_PQ_M,
-                    ENCODER_PARAMETER_PQ_CODE_COUNT_DEFAULT,
-                    v -> v > 0 && v < ENCODER_PARAMETER_PQ_CODE_COUNT_LIMIT
-                )
-            )
-            .addParameter(
-                ENCODER_PARAMETER_PQ_CODE_SIZE,
-                new Parameter.IntegerParameter(
-                    ENCODER_PARAMETER_PQ_CODE_SIZE,
-                    ENCODER_PARAMETER_PQ_CODE_SIZE_DEFAULT,
-                    v -> v > 0 && v < ENCODER_PARAMETER_PQ_CODE_SIZE_LIMIT
-                )
-            )
-            .setRequiresTraining(true)
-            .setMapGenerator(
-                ((methodComponent, methodComponentContext) -> MethodAsMapBuilder.builder(
-                    FAISS_PQ_DESCRIPTION,
-                    methodComponent,
-                    methodComponentContext
-                ).addParameter(ENCODER_PARAMETER_PQ_M, "", "").addParameter(ENCODER_PARAMETER_PQ_CODE_SIZE, "x", "").build())
+            .build()
+    );
+
+    // TODO: To think about in future: for PQ, if dimension is not divisible by code count, PQ will fail. Right now,
+    // we do not have a way to base validation off of dimension. Failure will happen during training in JNI.
+    // Define methods supported by faiss
```

Comment:
Let's create a github issue for this TODO and add it here.

Also, can you add details what failures customer will be seeing in the case dimensions are not divisible by code count.

Reply:
> Let's create a github issue for this TODO and add it here.

Will do

> Also, can you add details what failures customer will be seeing in the case dimensions are not divisible by code count.

Right, I think user will get some kind of generic message about training failing because we didnt want to send faiss error message in response. 

BTW this todo is not new.